tests/container_checker/test_container_checker.py

```diff
@@ -0,0 +1,259 @@
+    """Checks whether the stopped containers were started.
+
+    This function will check whether the stopped containers were restarted. If
+    the container was not restarted by the command 'config reload', then we
+    start it and check its status.
+
+    Args:
+        duthost: Hostname of DUT.
+        stopped_container_list: names of stopped containers.
+
+    Returns:
+        None.
+    """
+    for container_name in stopped_container_list:
+        logger.info("Checking the running status of container '{}'".format(container_name))
+        if is_container_running(duthost, container_name):
+            logger.info("Container '{}' is running.".format(container_name))
+        else:
+            logger.info("Container '{}' is not running and restart it...".format(container_name))
+            duthost.shell("sudo systemctl restart {}".format(container_name))
+            logger.info("Waiting until container '{}' is restarted...".format(container_name))
+            restarted = wait_until(CONTAINER_RESTART_THRESHOLD_SECS,
+                                   CONTAINER_CHECK_INTERVAL_SECS,
+                                   check_container_state, duthost, container_name, True)
+            if not restarted:
+                if is_hiting_start_limit(duthost, container_name):
+                    clear_failed_flag_and_restart(duthost, container_name)
+                else:
+                    pytest.fail("Failed to restart container '{}'".format(container_name))
+
+            logger.info("Container '{}' was restarted".format(container_name))
+
+
+def test_container_checker(duthosts, rand_one_dut_hostname, tbinfo):
+    """Tests the feature of container checker.
+
+    This function will check whether the container names will appear in the Monit
+    alerting message if they are stopped explicitly or they hit start limitation.
+
+    Args:
+        duthosts: list of DUTs.
+        rand_one_dut_hostname: hostname of DUT.
+        tbinfo: Testbed information.
+
+    Returns:
+        None.
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+    container_autorestart_states = duthost.get_container_autorestart_states()
+    disabled_containers = get_disabled_container_list(duthost)
+
+    bgp_neighbors = duthost.get_bgp_neighbors()
+    up_bgp_neighbors = [ k.lower() for k, v in bgp_neighbors.items() if v["state"] == "established" ]
+
+    skip_containers = disabled_containers[:]
+    # Skip 'radv' container on devices whose role is not T0.
+    if tbinfo["topo"]["type"] != "t0":
+        skip_containers.append("radv")
+
+    stopped_container_list = stop_containers(duthost, container_autorestart_states, skip_containers)
+    pytest_assert(len(stopped_container_list) > 0, "None of containers was stopped!")
+
+    # Wait for 6 minutes such that Monit has a chance to write alerting message into syslog.
+    logger.info("Sleep 6 minutes to wait for the alerting message...")
+    time.sleep(360)
```

Comment:
this seems very long? why do we need to wait for 6 minutes?

Reply:
Since the script `container_checker` was ran periodically by Monit and if one of containers was not running, Monit will write an alerting message into syslog after 5 minutes. So at here this pytest script has to wait for 6 minutes and then check whether the alerting message appeared in syslog or not.

Reply:
can you reduce monit duration?

Reply:
Currently the Monit configuration of script `container_checker` in image is as follows:
    
    check program container_checker with path "/usr/bin/container_checker"
        if status != 0 for 5 times within 5 cycles then alert repeat every 1 cycles

Can we change the configuration like this? Then I can reduce the waiting time at here.

    check program container_checker with path "/usr/bin/container_checker"
        if status != 0 for 2 times within 2 cycles then alert repeat every 1 cycles

Reply:
yes, change to 2 cycles in the test and change it back after test finishes.